zrml/court/src/lib.rs

```diff
@@ -1356,26 +1424,44 @@ mod pallet {
             let pool = <CourtPool<T>>::get();
             let pool_len = pool.len() as u32;
             let at_least_one_inflation_period =
-                |joined_at| now.saturating_sub(joined_at) >= T::InflationPeriod::get();
-            let total_stake = pool
-                .iter()
-                .filter(|pool_item| at_least_one_inflation_period(pool_item.joined_at))
-                .fold(0u128, |acc, pool_item| {
+                |join_at| now.saturating_sub(join_at) >= T::InflationPeriod::get();
+            let total_rewardable_stake = pool.iter().fold(0u128, |acc, pool_item| {
+                if at_least_one_inflation_period(pool_item.last_join_at) {
                     acc.saturating_add(pool_item.stake.saturated_into::<u128>())
-                });
-            if total_stake.is_zero() {
+                } else if at_least_one_inflation_period(pool_item.pre_period_join_at) {
+                    acc.saturating_add(pool_item.pre_period_join_stake.saturated_into::<u128>())
+                } else {
+                    acc
+                }
+            });
+            if total_rewardable_stake.is_zero() {
                 return T::WeightInfo::handle_inflation(0u32);
             }
 
             let mut total_mint = T::Currency::issue(inflation_period_mint);
 
-            for CourtPoolItem { stake, court_participant, joined_at, .. } in pool {
-                if !at_least_one_inflation_period(joined_at) {
+            for CourtPoolItem {
+                court_participant,
+                last_join_at,
+                stake,
+                pre_period_join_at,
+                pre_period_join_stake,
+                ..
+            } in pool
```

Comment:
Just so it does not get lost: Future optimization to avoid iterating over all pools twice is to cache the total stake.